dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPIImpl.java

```diff
@@ -69,17 +73,35 @@
 	final ContentletAPI contAPI;
 	final PermissionAPI perAPI;
 	private final IdentifierAPI identifierAPI;
+	private final FileAssetFactory fileAssetFactory;
+	private final ContentletCache contentletCache;
 
 	public FileAssetAPIImpl() {
-	    this(APILocator.getContentletAPI(),APILocator.getPermissionAPI(),APILocator.getSystemEventsAPI(),APILocator.getIdentifierAPI());
+	    this(
+	    		APILocator.getContentletAPI(),
+				APILocator.getPermissionAPI(),
+				APILocator.getSystemEventsAPI(),
+				APILocator.getIdentifierAPI(),
+				new FileAssetFactoryImpl(),
+				CacheLocator.getContentletCache()
+		);
 	}
 
-   public FileAssetAPIImpl(ContentletAPI contAPI,PermissionAPI perAPI, SystemEventsAPI systemEventsAPI, IdentifierAPI identifierAPI ) {
+	@VisibleForTesting
+    public FileAssetAPIImpl(
```

Comment:
I think you should implement a new method signature for this constructor and keep the old one. Internally the old constructor could call this new implementation

Reply:
I usually agree with backward compatibility, but I think people do not call the constructors, people use APILocator to delegate the construction, so I would say it is ok